gqlex.g4

```diff
@@ -71,13 +107,22 @@ conditionLogicalOperator
     ;
 
 conditionExpression
-    : lValue conditionOperator rValue (conditionLogicalOperator conditionExpression)?
+    : nameValue conditionOperator (nameValue | intValue)
+    | conditionExpression conditionLogicalOperator conditionExpression 
```

Comment:
Even if we ignore operator precedence for now, we should support parentheses in expressions.